.github/workflows/lint-code.yml

```diff
@@ -0,0 +1,51 @@
+name: Lint Python code
+
+on:
+  pull_request:
+    branches:
+      - "main"
+      - "release-**"
+    paths-ignore:
+      - "docs/**"
+      - "**.md"
+      - CODEOWNERS
+      - LICENSE
+
+permissions:
+  contents: read
+
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  pull-requests: read
+  # Optional: Allow write access to checks to allow the action to annotate code in the PR.
+  checks: write
+
+jobs:
+  code-linter:
+    name: "Run mypy linting"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'
```

Comment:
I think it is better if take the python version from our pyproject.toml file. You can change the python version in toml file to a 3.12.2:
```
      - name: Extract Python version from pyproject.toml
        id: python-version
        run: |
          echo "PYTHON_VERSION=$(cat pyproject.toml | grep '^python = ' | sed 's/python = \"//g' | sed 's/\"//g')" >> $GITHUB_ENV
      
      - name: Setup Python
        uses: actions/setup-python@v2
        with:
          python-version: ${{ env.PYTHON_VERSION }}
```

Reply:
done